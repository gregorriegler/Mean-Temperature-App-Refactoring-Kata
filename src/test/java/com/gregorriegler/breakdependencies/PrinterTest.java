package com.gregorriegler.breakdependencies;

import org.approvaltests.ApprovalUtilities;
import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.io.ByteArrayOutputStream;
import java.time.YearMonth;

public class PrinterTest {

    @Test
    void should_print() {
        ByteArrayOutputStream output = new ApprovalUtilities().writeSystemOutToStringBuffer();
        MeanTemperatureApplication printer = new MeanTemperatureApplication();
        printer.print(YearMonth.of(2020, 1), YearMonth.of(2020, 2), 1d);

        Approvals.verify(output);
    }
}
