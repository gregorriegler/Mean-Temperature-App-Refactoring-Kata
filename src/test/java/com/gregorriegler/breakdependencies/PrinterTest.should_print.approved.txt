2020-01 to 2020-02 mean temperature: 1.0 Â°C
